src/hevm/src/EVM/Expr.hs

```diff
@@ -521,7 +521,7 @@ indexWord (Lit idx) (JoinBytes zero        one        two       three
   | idx == 29 = twentynine
   | idx == 30 = thirty
   | idx == 31 = thirtyone
-  | otherwise = LitByte 0
+  | otherwise = undefined
```

Comment:
Lets use `error` instead here. Smth like this:

```suggestion
  | otherwise = error $ "Internal error: idx must be <= 31 (actual: " <> (show . num $ idx) <> ")"
```